killerbee/dev_cc253x.py

```diff
@@ -0,0 +1,272 @@
+        self.capabilities.setcapab(KBCapabilities.FREQ_2400, True)
+        self.capabilities.setcapab(KBCapabilities.SNIFF, True)
+        self.capabilities.setcapab(KBCapabilities.SETCHAN, True)
+        #self.capabilities.setcapab(KBCapabilities.INJECT, True)
+        #self.capabilities.setcapab(KBCapabilities.PHYJAM_REFLEX, True)
+        #self.capabilities.setcapab(KBCapabilities.SET_SYNC, True)
+
+    # KillerBee expects the driver to implement this function
+    def get_dev_info(self):
+        '''
+        Returns device information in a list identifying the device.
+        @rtype: List
+        @return: List of 3 strings identifying device.
+        '''
+        return [self.name, "CC253x", ""]
+
+    # KillerBee expects the driver to implement this function
+    def sniffer_on(self, channel=None):
+        '''
+        Turns the sniffer on such that pnext() will start returning observed
+        data.  Will set the command mode to Air Capture if it is not already
+        set.
+        @type channel: Integer
+        @param channel: Sets the channel, optional
+        @rtype: None
+        '''
+        self.capabilities.require(KBCapabilities.SNIFF)
+
+        if channel != None:
+            self.set_channel(channel)
+
+        # Enable power in 802.15.4 radio
+        self.dev.ctrl_transfer(CC253x.USB_DIR_OUT, CC253x.USB_POWER_ON, wIndex = 4)
+        while True:
+            # check if powered up
+            power_status = self.dev.ctrl_transfer(CC253x.USB_DIR_IN, CC253x.USB_POWER_STATUS, data_or_wLength = 1)
+            if power_status[0] == 4:
+                break
+            time.sleep(0.1)
+
+        # Set channel
+        self.dev.ctrl_transfer(CC253x.USB_DIR_OUT, CC253x.USB_XFER_CHAN, wIndex = 0, data_or_wLength = [self._channel])
+        self.dev.ctrl_transfer(CC253x.USB_DIR_OUT, CC253x.USB_XFER_CHAN, wIndex = 1, data_or_wLength = [0x00])
+
+        # Start capture
+        self.dev.ctrl_transfer(CC253x.USB_DIR_OUT, CC253x.USB_XFER_START)
+        self.__stream_open = True
+
+    # KillerBee expects the driver to implement this function
+    def sniffer_off(self):
+        '''
+        Turns the sniffer off, freeing the hardware for other functions.  It is
+        not necessary to call this function before closing the interface with
+        close().
+        @rtype: None
+        '''
+        self.dev.ctrl_transfer(CC253x.USB_DIR_OUT, CC253x.USB_XFER_STOP)
+        self.__stream_open = False
+
+    # KillerBee expects the driver to implement this function
+    def set_channel(self, channel):
+        '''
+        Sets the radio interface to the specifid channel (limited to 2.4 GHz channels 11-26)
+        @type channel: Integer
+        @param channel: Sets the channel, optional
+        @rtype: None
+        '''
+        self.capabilities.require(KBCapabilities.SETCHAN)
+
+        if channel >= 11 or channel <= 26:
+            self._channel = channel
+        else:
+            raise Exception('Invalid channel')
```

Comment:
In other implementations I think we change the channel here -- so that this can happen mid capture, etc -- without the developer using KillerBee to need to know that you need to stop, change the channel, and restart. Could we do the same here?